src/pieces/queen.ts

```diff
@@ -5,16 +5,90 @@ import Field from '../field';
 class Queen extends Piece {
     _display: string;
 
-    constructor(protected side: string) {
+    constructor(public side: string) {
         super(side);
         this._display = `<i class="fas fa-chess-queen ${side}"></i>`;
     }
 
     findLegalMoves(board: Board, actualField: Field): string[] {
         const possibleMoves: string[] = new Array();
-
+        const attacking = this.findAttackingMoves(board, actualField);
+        attacking.forEach(position => {
+            let x = Number(position![0]);
+            let y = Number(position![2]);
+            if (board.fields[x][y].piece === null) {
+                possibleMoves.push(position!);
+            } else {
```

Comment:
Ten else nie jest potrzebny, możemy zrobić dwa ify i to będzie lepsze rozwiazanie